src/py21cmfast/outputs.py

```diff
@@ -202,10 +202,18 @@ class PerturbedField(_OutputStructZ):
     ]
 
     def _get_box_structures(self) -> Dict[str, Union[Dict, Tuple[int]]]:
-        return {
-            "density": (self.user_params.HII_DIM,) * 3,
-            "velocity": (self.user_params.HII_DIM,) * 3,
-        }
+        if self.user_params.OUTPUT_ALL_VEL:
+            return {
+                "density": (self.user_params.HII_DIM,) * 3,
+                "velocity_x": (self.user_params.HII_DIM,) * 3,
+                "velocity_y": (self.user_params.HII_DIM,) * 3,
+                "velocity": (self.user_params.HII_DIM,) * 3,
+            }
+        else:
+            return {
+                "density": (self.user_params.HII_DIM,) * 3,
+                "velocity": (self.user_params.HII_DIM,) * 3,
+            }
```

Comment:
@IvanNikolic21  I added a bit out this to prevent outputting vx and vy when not needed. could you please test if this actually works?

Reply:
It works!

Reply:
so no vx and vy in the output perturb files when the flag is off? cool!